### ISSUE TITLE
Export canvas to png file on a windows machine

### ISSUE THREAD

**Issue by ivan-marroquin** [sandialabs/toyplot#204]
Hi,

Many thanks for making available this package.

I have Python 3.6.5 on a Windows 10 machine. I ran the pip installation command, below you have a part of the installation messages:

Successfully built toyplot reportlab
Installing collected packages: arrow, custom-inherit, multipledispatch, pypng, reportlab, toyplot
c:\temp\python\python3.6.5\lib\site-packages\setuptools\command\install.py:37: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
  setuptools.SetuptoolsDeprecationWarning,
c:\temp\python\python3.6.5\lib\site-packages\setuptools\command\install.py:37: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
  setuptools.SetuptoolsDeprecationWarning,
c:\temp\python\python3.6.5\lib\site-packages\setuptools\command\install.py:37: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
  setuptools.SetuptoolsDeprecationWarning,
c:\temp\python\python3.6.5\lib\site-packages\setuptools\command\install.py:37: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
  setuptools.SetuptoolsDeprecationWarning,
c:\temp\python\python3.6.5\lib\site-packages\setuptools\command\install.py:37: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
  setuptools.SetuptoolsDeprecationWarning,
c:\temp\python\python3.6.5\lib\site-packages\setuptools\command\install.py:37: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
  setuptools.SetuptoolsDeprecationWarning,
c:\temp\python\python3.6.5\lib\site-packages\setuptools\command\install.py:37: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
  setuptools.SetuptoolsDeprecationWarning,
Successfully installed arrow-1.2.2 custom-inherit-2.4.0 multipledispatch-0.6.0 pypng-0.0.21 reportlab-3.6.8 toyplot-1.0.1

I think that despite the warning messages, the installation went OK. Then, I installed the latest version of Ghostscript.

I opened a DOS command window, then I ran Python and tried these commands:
import numpy
import toyplot
import os
path= 'C:\\Users\\IMarroquin\\Documents\\My_Python_Scripts\\'
os.chdir(path)
x = numpy.linspace(0, 10)
y = x ** 2
canvas = toyplot.Canvas(width=300, height=300)
axes = canvas.cartesian()
mark = axes.plot(x, y)
toyplot.reportlab.png.render(canvas, 'toto.png')

I get the following error message:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'toyplot' has no attribute 'reportlab'

Any suggestions?

Ivan

**Comment by tshead2** [comment]
Ivan:

Thanks for reaching out ... it looks to me like you're just missing an import:

```import toyplot.reportlab```

... which should fix the problem.  Also, although your code should work fine, the slightly-better way to handle this would be:

```import toyplot.png
...
toyplot.png.render(canvas, 'toto.png')
```

Cheers,
Tim

**Comment by ivan-marroquin** [comment]
Hi Tim,

Thanks for your reply. I tried both commands:
toyplot.reportlab.png.render(canvas, 'toto.png')
toyplot.png.render(canvas, 'toto.png')

and both gave the same error message: 
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'toyplot.reportlab' has no attribute 'png' OR AttributeError: module 'toyplot' has no attribute 'png'

When I do the import of toyplot and toyplot.reportlab, I don't get any error message.

Any suggestions?

Ivan

**Comment by tshead2** [comment]
Ivan:

I just realized that my answer was misleading; in current versions of Toyplot the *only* way to export a PNG is with toyplot.png.render().  So do that :)

Cheers,
Tim

**Comment by ivan-marroquin** [comment]
Hi Tim,

Thanks for your collaboration. Unfortunately, I still get the same error message:

>>> toyplot.png.render(canvas, 'toto.png')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'toyplot' has no attribute 'png'

When I tried the Python code on VSC, I noticed that toyplot.reportlab. --> has a png module, the same for toyplot. --> has a png module.

So, I kept trying and I think the way to make it work:

import numpy
import toyplot
import toyplot.png #added this line
import os
path= 'C:\Users\IMarroquin\Documents\My_Python_Scripts\'
os.chdir(path)
x = numpy.linspace(0, 10)
y = x ** 2
canvas = toyplot.Canvas(width=300, height=300)
axes = canvas.cartesian()
mark = axes.plot(x, y)
toyplot.png.render(canvas, 'toto.png') #as per your suggestion

I attached the generated png file. Note that the axes are missing in display, is this the correct output format?

Ivan
[toyplot_output.zip](https://github.com/sandialabs/toyplot/files/9030444/toyplot_output.zip)
